ballerina/csv_stream.bal

```diff
@@ -33,7 +33,7 @@ public class CSVStream {
     # + return - A CSV record as a string array when a record is avaliable in the stream or
     # `()` when the stream reaches the end
     public isolated function next() returns record {| string[] value; |}|Error? {
-        var recordValue = readRecord(self.readableTextRecordChannel, COMMA);
```

Comment:
Instead of only using commas as the delimiter, the field separator should be derived from the channel.